docs/usage.md

```diff
@@ -29,6 +29,18 @@ system:
 sudo curl -L --output /usr/local/bin/pie https://github.com/php/pie/releases/latest/download/pie.phar && sudo chmod +x /usr/local/bin/pie
 ```
 
+### Docker installation
+
+PIE is published as binary-only Docker image, so you can install it easily during your Docker build:
+
+```Dockerfile
+COPY --form=ghcr.io/php/pie:latest /pie /usr/bin/pie
+```
+
+Instead of `latest` you can also use explicit versions like `x.y.z`, `x.y` or `x`, depending on stability level you want to achieve.
+
+If you want to verify signature of the PHAR, you need to additionally download `*.asc` file from [corresponding release](https://github.com/php/pie/releases).
```

Comment:
As part of my PRs #127 and #128 to do some preliminary GitHub Action cleanup, I've also tried to run pie.phar in a clean `php:8.3-cli` Docker container. This failed, because `ext-zip` is not installed by default.

Given that information, I'm wondering it shipping PIE as proposed in this PR is a value-add for the user. Perhaps the Docker image should instead contain a “PIE installer”, which:

1. `docker-php-ext-install zip` according to best practices (incl. the necessary cleanup of build dependencies) with some smartness to handle both Debian and Alpine images.
2. Places PIE into `/usr/bin/pie` after appropriately verifying the signature.

Usage would then look something like this:

```
COPY --from=ghcr.io/php/pie:latest /usr/bin/install-pie
RUN /usr/bin/install-pie
```

Alternatively the Docker image contain `pie.phar` embedded into a wrapper script that checks the prerequisites, installs them when necessary and then executes pie.phar.

Reply:
> These images are not meant to be used as PIE runtime, only as a interim carrier for easy `COPY --from` during custom Docker builds.

Yes, I understand that. I'm questioning the value of that. If as a user I need to think about the necessary steps to make PIE actually functional, then I might as well grab the pie.phar directly.

If use a Docker-optimized installation mechanism, then I expect it to simply work without any further action within a Docker context. 

> I am not sure if this should be PIE's responsibility to prepare runtime

It should not be PIE's responsibility, but the the responsibility of the distribution mechanism (i.e. the Docker image containing PIE). It can be as simple as a shell script like the following (untested):

```
#!/bin/bash

prepare_environment() {
	export DEBIAN_FRONTEND="noninteractive"
	savedAptMark="$(apt-mark showmanual)"
	buildDeps="
		libzip-dev
	"
	set -ex
	apt-get update
	apt-get install -y $buildDeps --no-install-recommends
	docker-php-ext-install -j "$(nproc)" zip
	apt-mark auto '.*' > /dev/null;
	apt-mark manual $savedAptMark;
	ldd "$(php -r 'echo ini_get("extension_dir");')"/*.so \
		| awk '/=>/ { so = $(NF-1); if (index(so, "/usr/local/") == 1) { next }; gsub("^/(usr/)?", "", so); print so }' \
		| sort -u \
		| xargs -r dpkg-query --search \
		| cut -d: -f1 \
		| sort -u \
		| xargs -r apt-mark manual
	apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false
	rm -rf /var/lib/apt/lists/*
}

if ! pie --version >/dev/null 2>&1; then
    prepare_environment
fi

exec pie "$@"
```

Reply:
Downloading PHAR in Docker build may require installing `curl` or `wget`, while copying PHAR from Docker image can be done natively and is cached on builder level (no more downloading on subsequent builds). It's also more stable as Docker image and binary's name and won't change (regardless of changes in repo and release process), while URL to the PHAR can. I am not against providing installer, but it shouldn't be blocker here, since installer would require PHAR file anyway. Binary-only approach works well in Composer, I believe it has value here too ☺️.

Reply:
> If as a user I need to think about the necessary steps to make PIE actually functional, then I might as well grab the pie.phar directly.

@TimWolla I understand what you're saying, but I think that's out of scope for this specific request. To alleviate this confusion, I think the image name we build here shouldn't be `ghcr.io/php/pie` but something more like `ghcr.io/php/pie-phar` or `ghcr.io/php/pie-bin`, to indicate that you can't actually "run" this image. Having a separate runnable Docker image with all the dependencies is something that could certainly be considered for addition, but I understand @Wirone's patch just provides that small benefit of `COPY --from=... /pie /usr/bin/pie` (I have personally used this technique a lot with Composer, so it is beneficial, IMO); I've also suggested documentation enhancements to make the fact you are responsible for installing the dependencies using this.

> Maybe we should add -bin suffix to make it explicit, just like it's done in Composer (then it's possible to provide fully usable images in the future without breaking naming convention).

@Wirone yes - I just caught this suggestion, and aligns with what I just wrote above :100: